wrappers/python/nntile/utils/constructors.py

```diff
@@ -39,38 +40,113 @@
 }
 
 
-def from_array(
+def empty(
+    shape: Sequence[int],
+    basetile_shape: Sequence[int] | None = None,
+    dtype: Tensor = Tensor_fp32,
+    mpi_distr: Sequence[int] | None = None,
+    next_tag: int = 0
+):
+    A_traits = TensorTraits(shape, basetile_shape or shape)
+    if mpi_distr is None:
+        mpi_distr = [0] * A_traits.grid.nelems
+    A_value = dtype(A_traits, mpi_distr, next_tag)
+    return A_value
+
+
+def empty_like(
     A: np.ndarray,
     basetile_shape: Sequence[int] | None = None,
-    mpi_distr=[0],
-    next_tag=0,
+    mpi_distr: Sequence[int] | None = None,
+    next_tag: int = 0
 ):
-    A_traits = TensorTraits(A.shape, basetile_shape or A.shape)
+    dtype = np2nnt_type_mapping[type(A.dtype)]
+    return empty(A.shape, basetile_shape, dtype, mpi_distr, next_tag)
 
-    A_value = np2nnt_type_mapping[type(A.dtype)](A_traits, mpi_distr, next_tag)
 
+def from_array(
+    A: np.ndarray,
+    basetile_shape: Sequence[int] | None = None,
+    mpi_distr: Sequence[int] | None = None,
+    next_tag: int = 0,
+):
+    A_value = empty_like(A, basetile_shape, mpi_distr, next_tag)
     A_value.from_array(A)
     return A_value
 
 
 def to_numpy(tensor_nnt):
     np_res = np.zeros(
-        tensor_nnt.shape, order="F", dtype=nnt2np_type_mapping[type(tensor_nnt)]
+        tensor_nnt.shape,
+        order="F",
+        dtype=nnt2np_type_mapping[type(tensor_nnt)]
     )
     tensor_nnt.to_array(np_res)
     return np_res
 
 
-def zeros(shape: Sequence[int], dtype=Tensor_fp32):
-    np_dtype = nnt2np_type_mapping[dtype]
-    return from_array(np.zeros(shape, dtype=np_dtype))
+def zeros(
+    shape: Sequence[int],
+    basetile_shape: Sequence[int] | None = None,
+    dtype: Tensor = Tensor_fp32,
+    mpi_distr: Sequence[int] | None = None,
+    next_tag: int = 0,
+):
+    A_value = empty(shape, basetile_shape, dtype, mpi_distr, next_tag)
+    clear_async(A_value)
```

Comment:
Do we have strong guarantee about clear_async doing actual filling with zeros?
Isn't it better to use explicit fill_async(0, A_value)

Reply:
'clear_async' actually calls memset(0) under the hood. It sets every byte to 0. So, there is a strong guarantee all integer, bool and floating point values are zeros of proper type